### PR TITLE
Milk-V Duo enhancements

### PR DESCRIPTION
**Pull request by kinsamanka**
This PR replaces the binary-only files provided by the **milkv-duo-bootfiles** recipe.

Improvements over the existing setup:
- mainline OpenSBI (v1.4)
- update u-boot environment variables via `uEnv.txt` file
- change devicetree (dtb) without recompiling the kernel
- add recipe for mainline kernel

### CODE REVIEW

**Review by kraj — CHANGES_REQUESTED** [review]
good work, needs few tweaks

File: recipes-bsp/milkv-duo-fsbl/milkv-duo-fsbl.bb [hunk 1]
```diff
@@ -17,6 +17,7 @@ SECURITY_LDFLAGS = ""
 do_compile[depends] += "opensbi:do_deploy"
 
 do_compile () {
+    unset LDFLAGS
```
Comment (kraj) [inline recipes-bsp/milkv-duo-fsbl/milkv-duo-fsbl.bb:20]: please fix tabs vs spaces

**Comment by kraj** [comment]
I see new commits being added to this pull, which is good, but please mark it as a `draft` PR until you are happy for me to consider it.

**Comment by kinsamanka** [comment]
The mainline kernel is the last enhancement for this PR, I will set it to draft to iron out any unforeseen bugs. Thanks

**Review by kraj — APPROVED** [review]
lgtm

**Comment by Mrzhaoqiankun** [comment]
hi Can you give a detailed tutorial

**Pull request merged**